hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteBufferedP.java

```diff
@@ -65,36 +89,49 @@ public boolean isCooperative() {
 
     private void flush() {
         //noinspection unchecked
-        map.putAll(buffer);
+        distributedObjectWrapper.putAll(buffer);
         buffer.clear();
     }
 
-    public static ProcessorMetaSupplier supplier(String mapName) {
-        return ProcessorMetaSupplier.of(new Supplier(mapName));
+    private static SerializableClientConfig serializableConfig(ClientConfig clientConfig) {
+        return clientConfig != null ? new SerializableClientConfig(clientConfig) : null;
     }
 
-    public static ProcessorMetaSupplier supplier(String mapName, ClientConfig clientConfig) {
-        SerializableClientConfig config = clientConfig != null ? new SerializableClientConfig(clientConfig) : null;
-        return ProcessorMetaSupplier.of(new Supplier(mapName, config));
+    static final class DistributedObjectWrapper {
+
+        private final DistributedObject distributedObject;
+
+        DistributedObjectWrapper(DistributedObject distributedObject) {
+            this.distributedObject = distributedObject;
+        }
+
+        void putAll(Map m) {
+            if (distributedObject instanceof IMap) {
+                ((IMap) distributedObject).putAll(m);
+            } else {
+                ((ICache) distributedObject).putAll(m);
+            }
+        }
     }
 
     private static class Supplier implements ProcessorSupplier {
 
         static final long serialVersionUID = 1L;
 
-        private final String name;
         private final SerializableClientConfig clientConfig;
+        private final Distributed.Function<HazelcastInstance, DistributedObjectWrapper> extract;
 
-        private transient IMap map;
+        private transient DistributedObjectWrapper distributedObjectWrapper;
         private transient HazelcastInstance client;
 
-        Supplier(String name, SerializableClientConfig clientConfig) {
-            this.name = name;
+        Supplier(SerializableClientConfig clientConfig,
+                 Distributed.Function<HazelcastInstance, DistributedObjectWrapper> extract) {
```

Comment:
should be `Distributed.Function<HazelcastInstance, Consumer<Map>>` as noted above